### PR TITLE
Fix for error with params

### PR DESCRIPTION
**Pull request by fgrande**
# Description

This would fix Issue #2439 
Looks like "params" property breaks the call, and this is filled with the params.

### CODE REVIEW

**Review by sanjai0py — COMMENTED** [review]
Can you please add more details on how this fix works?

**Comment by fgrande** [comment]
Consider those calls

https://spring-boot-test-aplication.herokuapp.com/api/users/findById/:id (set 16 as :id)
https://spring-boot-test-aplication.herokuapp.com/api/users/findById/16

The difference between them are the "params" tag in the request.

**_Request with params :id :_** 

mode: 'none',
  method: 'GET',
  url: 'https://spring-boot-test-aplication.herokuapp.com/api/users/findById/16',
  headers: {},
  params: [
    {
      name: 'id',
      value: '16',
      enabled: true,
      type: 'path',
      uid: '0Ndm3TH0VGhdOE0CgP4f6'
    }
  ],
  responseType: 'arraybuffer',
  script: {},


**_Request without param :_**

 mode: 'none',
  method: 'GET',
  url: 'https://spring-boot-test-aplication.herokuapp.com/api/users/findById/16',
  headers: {},
  params: [],
  responseType: 'arraybuffer',
  script: {},
  vars: {},
  assertions: [],


As you can see the difference lays in the params property. Since it is already decoded (the request.url is - at this point - already filled with the correct param) we don't need it anymore. And putting it, axios refuse the request because of an exception related to the actual url composed :

 _header: 'GET /api/users/findById/16?0[name]=id&0[value]=16&0[enabled]=true&0[type]=path&0[uid]=w4AGcAZKgr0VVwXrnBKt7 HTTP/1.1\r\n' +
      'Accept: application/json, text/plain, */*\r\n' +
      'request-start-time: 1718263487771\r\n' +
      'User-Agent: axios/1.6.7\r\n' +
      'Accept-Encoding: gzip, compress, deflate, br\r\n' +
      'Host: spring-boot-test-aplication.herokuapp.com\r\n' +
      'Connection: keep-alive\r\n' +
      '\r\n',

So, removing the params property, works fine. Maybe is not the cleaner solution, but I can give the idea of what's happening.

Regards
FabioG

**Comment by fgrande** [comment]
Adding test files (related to the issue) 
[BrunoTest.zip](https://github.com/user-attachments/files/15815507/BrunoTest.zip)

File: packages/bruno-electron/src/ipc/network/index.js [hunk 1]
```diff
@@ -499,6 +499,8 @@ const registerNetworkIpc = (mainWindow) => {
       let response, responseTime;
       try {
         /** @type {import('axios').AxiosResponse} */
+        // Looks like the "params" create problems if filled
+        request.params = [];
```
Comment (sanjai0py) [inline packages/bruno-electron/src/ipc/network/index.js]: After interpolation, we can remove the params values as they are already interpolated in the URL, making the approach simpler.

https://github.com/usebruno/bruno/blob/9b382fa0911536a0bfa27ae0d8a67007cc1364f7/packages/bruno-electron/src/ipc/network/interpolate-vars.js#L118

`  delete request.params;
`

**Review by sanjai0py — APPROVED** [review]

File: packages/bruno-electron/src/ipc/network/index.js (hunk 1 above)
Comment (fgrande) [reply packages/bruno-electron/src/ipc/network/index.js]: @sanjai0py Thanks a lot for the hint (I'm still working on my knowledge of Bruno), I've just updated my PR. Let me know if it's ok, hope to be (somehow) helpful

**Comment by helloanoop** [comment]
Thanks @fgrande for investigating the issue and coming up with the fix.

I feel the code related to deleting the path params is better to be kept inside the `configureRequest` instead of having it in interpolate vars function. I have asked @sanjai0py to make this change right away in a separate PR.

Closing this PR. Thanks again for your work on this.

**Pull request closed without merging**